optimizely/helpers/condition.py

```diff
@@ -12,115 +12,180 @@
 # limitations under the License.
 
 import json
+import numbers
 
+from six import string_types
 
-class ConditionalOperatorTypes(object):
+from .validator import is_finite_number, are_values_same_type
+
+
+class ConditionOperatorTypes(object):
   AND = 'and'
   OR = 'or'
   NOT = 'not'
 
 
-DEFAULT_OPERATOR_TYPES = [
-  ConditionalOperatorTypes.AND,
-  ConditionalOperatorTypes.OR,
-  ConditionalOperatorTypes.NOT
-]
+class ConditionMatchTypes(object):
+  EXACT = 'exact'
+  EXISTS = 'exists'
+  GREATER_THAN = 'gt'
+  LESS_THAN = 'lt'
+  SUBSTRING = 'substring'
+
 
+class CustomAttributeConditionEvaluator(object):
+  """ Class encapsulating methods to be used in audience leaf condition evaluation. """
 
-class ConditionEvaluator(object):
-  """ Class encapsulating methods to be used in audience condition evaluation. """
+  CUSTOM_ATTRIBUTE_CONDITION_TYPE = 'custom_attribute'
 
   def __init__(self, condition_data, attributes):
     self.condition_data = condition_data
-    self.attributes = attributes
+    self.attributes = attributes or {}
 
-  def evaluator(self, condition):
-    """ Method to compare single audience condition against provided user data i.e. attributes.
+  def is_value_valid_for_exact_conditions(self, value):
+    """ Method to validate if the value is valid for exact match type evaluation
 
     Args:
-      condition: Integer representing the index of condition_data that needs to be used for comparison.
+      value: Value to validate
```

Comment:
nit. Period (.) at the end of line. Please make sure to do so in all messages in this PR.